### PR TITLE
Import goreleaser cross

### PR DESCRIPTION
**Pull request by gsoltis**
This is an unmodified import of https://github.com/goreleaser/goreleaser-cross into the `buildcontainer` directory.

The intention is that we will then modify it to build `turbo`'s cross-compilation images. Relevant additions have been upstreamed already, the remaining work is removing platforms we don't care about and adding tools that we use in addition to Go and goreleaser.

### CODE REVIEW

File: buildcontainer/.github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+#          cosign sign --key /tmp/cosign.key ghcr.io/goreleaser/goreleaser-cross:${{env.RELEASE_TAG}}
+#        env:
+#          COSIGN_KEY: ${{secrets.COSIGN_PRIVATE_KEY}}
+#          COSIGN_PASSWORD: ${{secrets.COSIGN_PASSWORD}}
+      - name: Generate changelog
+        run: make gen-changelog
+      - name: Release
+        uses: softprops/action-gh-release@v1
+        with:
+          body_path: changelog.md
```
Comment (mehulkar) [inline buildcontainer/.github/workflows/release.yaml:99]: Hope Github doesn't care about a `.github` directory nested inside?

**Review by mehulkar — COMMENTED** [review]
Curious, why isn't it possible to use this as a dependency and upstream the hooks and/or config we need to customize? Is it common to inline? Do we need to pull in upstream changes in the future?

**Comment by gsoltis** [comment]
@mehulkar I think the projects have different, but related goals. Goreleaser-cross aims to compile Go code on as many platforms as possible. We aim to compile Go code on 6 platforms, plus do some other stuff (javascript, rust, etc.).

I'm happy to upstream anything related to Go, and likewise pull in from upstream anything related to compiling Go, but I don't expect our version to stay Go-focused. As such, this is mostly a starting point for us, rather than something we'll be able to just add a customization layer to.

**Comment by mehulkar** [comment]
@gsoltis ok sounds good! For the purpose of this PR, do you want a review of the actual code, or can we assume that it's good as-is?

**Comment by gsoltis** [comment]
My intention is to make no changes in this PR so that we have a clean import. Anything that we would change as the result of a review will come in a later PR anyways.

**Comment by gsoltis** [comment]
@mehulkar for reference, here's my first batch of turbo-specific changes https://github.com/vercel/turborepo/pull/1930

**Review by mehulkar — APPROVED** [review]
👍🏾 🚢

**Comment by mehulkar** [comment]
This might be a good candidate for `git merge --no-ff` btw so we can keep `git log` for these files (thanks for preserving all the history btw!)

**Pull request merged**